pkg/dogstatsd/parse_events.go

```diff
@@ -68,24 +68,38 @@ func splitHeaderEvent(message []byte) ([]byte, []byte, error) {
 }
 
 func parseHeader(rawHeader []byte) (eventHeader, error) {
+	// Basic sanity check on the header (minimum length)
 	if len(rawHeader) < 7 {
 		return eventHeader{}, fmt.Errorf("invalid event header: %q", rawHeader)
 	}
+
+	// Extract out the raw title and text lengths
 	rawLengths := rawHeader[3 : len(rawHeader)-1]
 	sepIndex := bytes.Index(rawLengths, commaSeparator)
 	if sepIndex == -1 {
 		return eventHeader{}, fmt.Errorf("invalid event header: %q", rawHeader)
 	}
+
 	rawTitleLength := rawLengths[:sepIndex]
 	rawTextLength := rawLengths[sepIndex+1:]
+
+	// Convert title length to workable type and do a basic validity check on value
 	titleLength, err := parseInt64(rawTitleLength)
-	if err != nil {
+	if err != nil || titleLength < 0 {
 		return eventHeader{}, fmt.Errorf("invalid event header: %q", rawHeader)
 	}
+
+	// Ensure that title isn't empty
+	if titleLength == 0 {
+		return eventHeader{}, fmt.Errorf("invalid event: empty title")
+	}
+
+	// Convert text length to workable type and do a basic validity check on value
```

Comment:
Wouldn't that make sense to validate that `textLength` is not equal to `0` as well, the same way you did for `titleLength`? It seems that the text field is mandatory in events.

Reply:
Nevermind, we accept empty text for events.

Reply:
Yeah - the original title length check was just moved from the lower part of the diff so I assumed (and according to the slack convo and [issue](https://github.com/DataDog/datadog-agent/pull/6662) that the assumption is correct) that it is explicitly allowed.